### PR TITLE
Don't try to autodetect headers for tecio in system

### PR DESCRIPTION
**Pull request by lindsayad**
Refs #2896 

I thought about being more ambitious and tackling some other packages that toss in `/usr/include` and I ran into eigen...which we use a lot more, and so I got nervous and decided to go with the low-hanging fruit.

### CODE REVIEW

**Review by jwpeterson — APPROVED** [review]
Looks fine to me, assuming that it fixes the original issue you ran into.

**Comment by lindsayad** [comment]
Submitted fix for MOOSE failures at https://github.com/idaholab/moose/pull/17554

**Comment by lindsayad** [comment]
@GiudGiud confirmed that this fixed his build. I'll wait until the MOOSE fix gets merged and propagated and then I'll invalidate this target for the full green effect

**Pull request merged**
